lib/screens/create_region_screens/choose_region_id/choose_region_id.dart

```diff
@@ -15,54 +16,65 @@ class ChooseRegionId extends StatelessWidget {
-                      ]))));
+    return BlocConsumer<CreateRegionBloc, CreateRegionState>(
+      listenWhen: (_, current) => current.pageCommand != null,
+      listener: (context, state) {
+        final pageCommand = state.pageCommand;
 
-        default:
-          return const SizedBox.shrink();
-      }
-    });
+        if (pageCommand is ShowErrorMessage) {
+          eventBus.fire(ShowSnackBar(pageCommand.message));
+        }
+        BlocProvider.of<CreateRegionBloc>(context).add(const ClearCreateRegionPageCommand());
+      },
+      builder: (context, state) {
+        return WillPopScope(
+          onWillPop: () async {
+            BlocProvider.of<CreateRegionBloc>(context).add(const OnBackPressed());
+            return false;
+          },
+          child: Scaffold(
+            appBar: AppBar(
+                leading:
+                    BackButton(onPressed: () => BlocProvider.of<CreateRegionBloc>(context).add(const OnBackPressed())),
+                title: Text(context.loc.createRegionSelectRegionAppBarTitle)),
+            body: SafeArea(
+              minimum: const EdgeInsets.all(horizontalEdgePadding),
+              child: Stack(
+                children: [
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Center(
+                          child: Container(
+                              width: 60,
+                              height: 60,
+                              child: AuthenticationStatus(authenticationIdState: state.regionIdAuthenticationState))),
+                      const SizedBox(height: 10),
+                      TextFormFieldCustom(
+                          errorText: state.regionIdErrorMessage,
+                          initialValue: state.regionId,
+                          maxLength: 8,
+                          suffixText: ".rgn",
+                          autofocus: true,
+                          labelText: context.loc.createRegionChooseRegionIdInputFormTitle,
+                          onChanged: (val) => BlocProvider.of<CreateRegionBloc>(context).add(OnRegionIdChange(val))),
+                      const SizedBox(height: 20),
+                      Text(context.loc.createRegionChooseRegionIdDescription,
+                          style: Theme.of(context).textTheme.subtitle2OpacityEmphasis)
+                    ],
+                  ),
+                  Align(
+                      alignment: Alignment.bottomCenter,
+                      child: FlatButtonLong(
+                          enabled: state.regionIdAuthenticationState == AuthenticationState.valid,
+                          title: "${context.loc.createRegionSelectRegionButtonTitle} (3/5)",
+                          onPressed: () => BlocProvider.of<CreateRegionBloc>(context).add(const OnNextTapped())))
```

Comment:
i think some commas here will make this code look better

Reply:
fix, thank you it looks better